openml/runs/trace.py

```diff
@@ -346,9 +348,35 @@ def trace_from_xml(cls, xml):
 
         return cls(run_id, trace)
 
+    @classmethod
+    def merge_traces(cls, traces: List['OpenMLRunTrace']) -> 'OpenMLRunTrace':
+
+        merged_trace = OrderedDict()  # type: OrderedDict[Tuple[int, int, int], OpenMLTraceIteration]  # noqa E501
+
+        previous_iteration = None
+        for trace in traces:
+            for iteration in trace:
+                key = (iteration.repeat, iteration.fold, iteration.iteration)
+                if previous_iteration is not None:
+                    if (
+                        list(merged_trace[previous_iteration].parameters.keys())
+                        != list(iteration.parameters.keys())
+                    ):
+                        raise ValueError(
+                            'Cannot merge traces because the parameters are not equal: {} vs {}'.
+                            format(
+                                list(merged_trace[previous_iteration].parameters.keys()),
+                                list(iteration.parameters.keys()),
+                            )
+                        )
+                merged_trace[key] = iteration
+                previous_iteration = key
+
+        return cls(None, merged_trace)
```

Comment:
Thanks for the suggestion. I'm not 100% sure whether this will allow the same flexibility, for example, the first statement would only check the iterations in one trace, and it would be hard to extract a useful error message out of it. Therefore, i suggest leaving the code as it is.

Reply:
Right, I missed that.
```python
all_iterations = [iter for trace in traces for iter in trace]
for (iter1, iter2) in zip(all_iterations, all_iterations[1:]):
   if sorted(iter1.parameters) != sorted(iter2.parameters):
       raise ValueError(...)

```
should then work fine? But I am fine leaving this as is (and perhaps refactoring it later if we can find something we all like).